client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/design-picker-design-title.tsx

```diff
@@ -30,10 +44,20 @@ const DesignPickerDesignTitle: FC< Props > = ( { designTitle, selectedDesign } )
 		)
 	);
 
-	const bundleSettings = useBundleSettingsByTheme( selectedDesign.slug );
-
 	let badge: React.ReactNode = null;
-	if ( selectedDesign.software_sets && selectedDesign.software_sets.length > 0 ) {
+	if ( isEnabled( 'themes/tiers' ) ) {
+		const isDefaultVariation = isDefaultGlobalStylesVariationSlug( selectedStyleVariation?.slug );
+		const isLockedStyleVariation =
+			( ! selectedDesign.is_premium && shouldLimitGlobalStyles && ! isDefaultVariation ) ?? false;
```

Comment:
I copied this from:
https://github.com/Automattic/wp-calypso/blob/483892a5f840debe021b0395d732f8f0a935a713/packages/design-picker/src/components/unified-design-picker.tsx#L176-L177 
but there's a slightly different definition that doesn't check whether the theme is premium:
https://github.com/Automattic/wp-calypso/blob/4527c48fb086a38e17129da2125866914e2217b7/client/components/theme/index.jsx#L317-L319

@mmtr, since both of them where added by you, could you tell me the difference?

Thanks!

Reply:
Nice finding!

Styles variation of premium styles shouldn't trigger any Limited GS warning/upsell because to use a premium theme you need a Explorer plan which already gives you access to the GS style variations.

In the Design Picker (`unified-design-picker.tsx`), we check whether the theme is premium to ensure that users previewing a style variation of a premium theme still see the premium theme upsell instead of the GS upsell:

Previewing a style variation of a free theme in the DP | Previewing a style variation of a premium theme in the DP
--- | ---
GS upsell<img width="401" alt="Screenshot 2024-01-18 at 13 38 01" src="https://github.com/Automattic/wp-calypso/assets/1233880/84cbcd99-cd37-4353-ad2c-85022e73bfdb"> | Premium theme upsell<img width="511" alt="Screenshot 2024-01-18 at 13 38 24" src="https://github.com/Automattic/wp-calypso/assets/1233880/24183305-dd54-46ed-b85c-5ad3c1aa8915">

That fact that we're not doing the same in the Theme Showcase (`components/theme/index.jsx`) is causing a bug so we always show the GS upsell regardless of whether the theme is premium or free:

Previewing a style variation of a free theme in the TS | Previewing a style variation of a premium theme in the TS
--- | ---
GS upsell<img width="592" alt="Screenshot 2024-01-18 at 13 43 10" src="https://github.com/Automattic/wp-calypso/assets/1233880/5efd4891-be4a-4737-96f6-ad0469cbb722">
 | Premium theme upsell<img width="611" alt="Screenshot 2024-01-18 at 13 43 03" src="https://github.com/Automattic/wp-calypso/assets/1233880/ffa63d24-037f-4252-acbe-421c67db69f5">

We should show the premium theme upsell in the TS when previewing a style variation of a premium theme.

Reply:
Ok, so the code in [wp-calypso/client/components/theme/index.jsx](https://github.com/Automattic/wp-calypso/blob/4527c48fb086a38e17129da2125866914e2217b7/client/components/theme/index.jsx#L317-L319) needs to be fixed, right? I can do that on a followup PR.

Reply:
Fix PR: https://github.com/Automattic/wp-calypso/pull/86654